### PR TITLE
Dockerfile: add support for optional build against OracleSpatial client library

### PR DESCRIPTION
**Pull request by rouault**
Procedure:
1. Download instantclient-basic-linux.x64-19.8.0.0.0dbru.zip and
   instantclient-sdk-linux.x64-19.8.0.0.0dbru.zip from
   https://www.oracle.com/database/technologies/instant-client/linux-x86-64-downloads.html
   and put there under the instantclient/ directory

2. "make build WITH_ORACLE=ON"

If there's interest, I've the backport of this for the 7.6 branch

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,24 +21,48 @@ RUN git clone ${MAPSERVER_REPO} --branch=${MAPSERVER_BRANCH} --depth=100 /src
...
+       apt-get clean && \
+       rm -rf /var/lib/apt/lists/* && \
+       mkdir -p /opt/instantclient && \
+       cd /opt/instantclient && \
+       (for i in /tmp/instantclient/*.zip; do bsdtar --strip-components=1 -xvf $i; done) && \
+       ls -ltr /opt/instantclient && \
+       ln -s libnnz19.so /opt/instantclient/libnnz18.so; \
+     else \
+       mkdir -p /opt/instantclient/sdk && \
+       touch /opt/instantclient/dummy.txt; \
```
Comment (sbrunner) [inline Dockerfile:40]: Why do we need this file?

**Review by sbrunner — COMMENTED** [review]
Just one question, also that looks good :-)

File: Dockerfile (hunk 1 above)
Comment (pmauduit) [reply Dockerfile:40]: also  the "ln -s" statement is surprising :) (version 19 -> 18)

Comment (rouault) [reply Dockerfile]: I suspected that the "COPY --from=builder /opt/instantclient /usr/local/lib/" might fail, but it does not. I've just forced pushed without this line

Comment (rouault) [reply Dockerfile]: > also the "ln -s" statement is surprising :) (version 19 -> 18)

this is to workaround an issue in MapServer detection of Oracle SDK. Adressed upstream per https://github.com/MapServer/MapServer/pull/6146

Comment (pmauduit) [reply Dockerfile]: ok makes sense

**Comment by sbrunner** [comment]
@rouault I get an error in the CI
https://github.com/camptocamp/docker-mapserver/pull/60/checks?check_run_id=1161031708#step:5:4797
It's a known issue in Docker, you should add a layer before... (or trying to use ubuntu-20.04)

**Comment by rouault** [comment]
> It's a known issue in Docker, you should add a layer before... (or trying to use ubuntu-20.04)

what is the issue exactly ? that "COPY --from=builder /opt/instantclient /usr/local/lib/" results in an empty layer. Works fine here on ubuntu 16.04...
Perhaps in the builder part I should just copy files from /opt/instantclient to /usr/local/lib, instead of doing this in the runner ?

**Comment by sbrunner** [comment]
The Docker issue is here: https://github.com/moby/moby/issues/37965
And the first message contains as I know the better workaround :-)

**Comment by rouault** [comment]
I've revised/simplified the way to deal with Oracle SDK to avoid this Docker issue

**Comment by sbrunner** [comment]
@rouault Thanks :-)

**Review by sbrunner — APPROVED** [review]

**Pull request merged**